zap/src/main/java/org/parosproxy/paros/core/scanner/VariantMultipartFormParameters.java

```diff
@@ -55,10 +55,11 @@
     // insensitive & DOTALL, and hit "test")
 
     private List<NameValuePair> params = Collections.emptyList();
-    private List<MultipartFormParameter> multiPartParams = new ArrayList<MultipartFormParameter>();
+    private List<MultipartFormParameter> multiPartParams;
 
     @Override
     public void setMessage(HttpMessage msg) {
+        multiPartParams = new ArrayList<MultipartFormParameter>();
```

Comment:
@thc202  @kingthorin my initial thought is to call setMessage again after setting the message for parsing the message again. This can be done from the setParameter api or by exposing a new api which can handle multiple updates and hence the new Api will not impact the performance of existing setparameter api.
WDYT ?

Reply:
It's not necessary to re-parse the message to be able to set multiple parameters.

Reply:
sure will think more on this.